src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAttribute.cs

```diff
@@ -29,127 +29,153 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Security.Claims;
-using Microsoft.IdentityModel.Logging;
+using static Microsoft.IdentityModel.Logging.LogHelper;
 
 namespace Microsoft.IdentityModel.Tokens.Saml
 {
+    /// <summary>
+    /// Represents the Attribute element.
+    /// </summary>
     public class SamlAttribute
     {
         private string _name;
         private string _nameSpace;
-        private Collection<string> _attributeValues = new Collection<string>();
         private string _originalIssuer;
         private string _attributeValueXsiType = ClaimValueTypes.String;
         private List<Claim> _claims;
         private string _claimType;
 
         // TODO remove this internal
+        /// <summary>
+        /// Initializes a new instance of the SamlAttribute class.
+        /// </summary>
         internal SamlAttribute()
         {
+            Values = new List<string>();
```

Comment:
This is not needed. It is set twice below, in the constructor and on the property.